### PR TITLE
API Harmonization

### PR DESCRIPTION
**Pull request by theosirian**
This PR as discussed internally tries to harmonize all siwe libraries to a similar behavior across all different languages.

- [x] API uniformization across libraries
- [x] Improve message syntax checking
- [x] Improve error messages
- [x] Improve tests and adding siwe-ts test vectors

### CODE REVIEW

File: native/siwe_ex/src/lib.rs [hunk 1]
```diff
@@ -174,8 +186,8 @@ rustler::init!(
     [
         parse,
         to_str,
-        validate_sig,
-        validate,
+        verify_sig,
+        verify,
```
Comment (awoie) [inline native/siwe_ex/src/lib.rs:190]: can we bump version number of siwe-ex?

File: test/siwe_test.exs [hunk 2]
```diff
@@ -2,7 +2,7 @@ defmodule SiweTest do
   use ExUnit.Case
   doctest Siwe
 
-  test "helper functions are equivilant" do
+  test "helper functions are equivalent" do
```
Comment (awoie) [inline test/siwe_test.exs:5]: should we use the global test vectors at some point?

**Review by awoie — CHANGES_REQUESTED** [review]
lgtm except version bump and global test cases not being used but the latter can be done in a separate PR.

**Comment by krhoda** [comment]
References to `verify_time` should be removed and the tests changed from their existing format to one that uses the test vectors in the main repo. Otherwise, looks good to me.

**Comment by krhoda** [comment]
Definitely still needs `siwe-rs` PR to come through first so we can point the `Cargo.toml` away from the local filesystem.

I've updated the tests to match the API changes, but I found issues passing optional parameters to `verify`, I'll try to change things so we use a struct which I think will solve our problems.

**Comment by krhoda** [comment]
The tests are now updated, the strange thing is that the `siwe-rs`'s `verify` function seems to be accepting a bad signature as long as everything else is valid, as seen by the commented out `assert` in the tests, I want to understand why that is happening before calling it all good.

**Comment by krhoda** [comment]
`verify` doesn't seem to reject bad messages from my testing, and it seems like it might be an underlying issue with `siwe-rs`, I have the latest `api-harmonization` branch but it still isn't working as expected.

**Comment by krhoda** [comment]
Got the tests working with the new API, seems great to me. Just need to publish the changes to the rust crate, then we can change this repo's Cargo.toml to no longer point at the local fs.

**Review by krhoda — CHANGES_REQUESTED** [review]
Just get the `Cargo.toml` pointed a crate and not the local fs for `siwe` dep.

**Comment by krhoda** [comment]
Looks great to me, tests work locally and no longer uses fs for `siwe` dep!

**Review by krhoda — APPROVED** [review]

**Pull request merged**